Add additional LLM span attributes

karthikscale3 opened the pull request:
Fixes #

## Changes

Please provide a brief description of the changes here.

Adding additional attributes to LLM span attributes.

Note: if the PR is touching an area that is not listed in the [existing areas](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md), or the area does not have sufficient [domain experts coverage](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/CODEOWNERS), the PR might be tagged as [experts needed](https://github.com/open-telemetry/semantic-conventions/labels/experts%20needed) and move slowly until experts are identified.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,30 @@ groups:
+        type: double
+        brief: The top_k sampling setting for the LLM request.
+        examples: [1.0]
+        tag: llm-generic-request
```
not needed (please remove from other attributes introduced in this PR)
```suggestion
```

karthikscale3 replied:
thanks! done

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,30 @@ groups:
+        brief: The top_k sampling setting for the LLM request.
+        examples: [1.0]
+        tag: llm-generic-request
+      - id: request.stop
```
```suggestion
      - id: request.stop_sequence
```

or 

```suggestion
      - id: request.stop_sequences
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,30 @@ groups:
+        tag: llm-generic-request
+      - id: request.stop
+        stability: experimental
+        type: double
```
```suggestion
        type: string[]
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,30 @@ groups:
+      - id: request.stop
+        stability: experimental
+        type: double
+        brief: The stop sequences to provide.
```
```suggestion
        brief: The stop sequences provided in the request.
```

lmolkova commented on .chloggen/839.yaml:
```diff
@@ -0,0 +1,4 @@
+change_type: enhancement
+component: gen-ai
+note: Adding top_k and stop to the list of required attributes for the gen_ai.request.model event.
```
```suggestion
note: Adding `gen_ai.request.top_k`, `gen_ai.request.presence_penalty`,  `gen_ai.request.frequency_penalty` and `gen_ai.request.stop_sequences` attributes.
```

karthikscale3 commented:
thanks @lmolkova . I have addressed all your review comments.

lmolkova reviewed (approved):
yaml changes look good, but please make sure to regenerate markdown tables

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,26 @@ groups:
+        stability: experimental
+        type: double
+        brief: The presence penalty to provide.
+        examples: ['0.1']
```
```suggestion
        examples: [0.1]
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,26 @@ groups:
+        stability: experimental
+        type: double
+        brief: The frequency penalty to provide.
+        examples: ['0.1']
```
```suggestion
        examples: [0.1]
```

nirga commented on docs/attributes-registry/gen-ai.md:
```diff
@@ -19,6 +19,10 @@ This document defines the attributes used to describe telemetry in the context o
 | `gen_ai.request.model`           | string   | The name of the LLM a request is being made to.                                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.temperature`     | double   | The temperature setting for the LLM request.                                                     | `0.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.top_p`           | double   | The top_p sampling setting for the LLM request.                                                  | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gen_ai.request.stop`           | string   | Up to 4 sequences where the API will stop generating further tokens.                             | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
There's a mismatch in the name between this and the yaml. You should generate this automatically

karthikscale3 replied:
Got it! Will generate and push this back up

nirga commented on docs/attributes-registry/gen-ai.md:
```diff
@@ -19,6 +19,10 @@ This document defines the attributes used to describe telemetry in the context o
 | `gen_ai.request.model`           | string   | The name of the LLM a request is being made to.                                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.temperature`     | double   | The temperature setting for the LLM request.                                                     | `0.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.top_p`           | double   | The top_p sampling setting for the LLM request.                                                  | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gen_ai.request.stop`           | string   | Up to 4 sequences where the API will stop generating further tokens.                             | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
The comment is not clear to me. First, it says "4", but for example [cohere](https://docs.cohere.com/reference/chat) allows 5 sequences. Second, the example isn't right - it's a sequence so it should be a string, not a number. Third, type should be an array of strings IMO

karthikscale3 replied:
Good catch! Thanks @nirga, I will fix and push these up

drewby reviewed (commented):
@karthikscale3, can you resolve the conversations (including two changes from Liudmila and we can approve and merge)

joaopgrassi commented on model/registry/gen-ai.yaml:
```diff
@@ -45,6 +45,26 @@ groups:
+        type: string[]
+        brief: The stop sequences provided in the request.
+        examples: ['\n']
+      - id: request.frequency_penalty
```
Would this be a candidate for future attributes under `frequency`? Also the one below with `presence`? Should we reserve these namespaces already?

marcalff commented on model/trace/gen-ai.yaml:
```diff
@@ -23,6 +23,14 @@ groups:
+        requirement_level: recommended
+      - ref: gen_ai.request.stop
+        requirement_level: recommended
+      - ref: gen_ai.request.frequncy_penalty
```
Typo, which breaks CI.

karthikscale3 commented:
> @karthikscale3, can you resolve the conversations (including two changes from Liudmila and we can approve and merge)

Apologies for the delay. Updated the PR
- incorporated feedbacks and resolved comments
- regenerated markdowns

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -54,6 +54,29 @@ groups:
+        type: double
+        brief: The presence penalty setting for the LLM request.
+        examples: [0.1]
+        tag: llm-generic-request
```
```suggestion
```
(not needed )

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -54,6 +54,29 @@ groups:
+        type: double
+        brief: The frequency penalty setting for the LLM request.
+        examples: [0.1]
+        tag: llm-generic-request
```
```suggestion
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -54,6 +54,29 @@ groups:
+      - id: request.top_k
+        stability: experimental
+        type: double
+        brief: The top_k sampling setting for the LLM request.
```
```suggestion
        brief: The top_k sampling setting for the GenAI request.
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -54,6 +54,29 @@ groups:
+      - id: request.frequency_penalty
+        stability: experimental
+        type: double
+        brief: The frequency penalty setting for the LLM request.
```
```suggestion
        brief: The frequency penalty setting for the GenAI request.
```

lmolkova commented on model/registry/gen-ai.yaml:
```diff
@@ -54,6 +54,29 @@ groups:
+      - id: request.presence_penalty
+        stability: experimental
+        type: double
+        brief: The presence penalty setting for the LLM request.
```
```suggestion
        brief: The presence penalty setting for the GenAI request.
```

lmolkova reviewed (approved):
Left some cosmetic comment, LGTM otherwise

karthikscale3 commented:
> Left some cosmetic comment, LGTM otherwise

Thanks @lmolkova! I have updated the PR again. Looking for one more approval and then we should be good to merge this one. cc @drewby

nirga reviewed (approved):
@lmolkova I think this is good to go

nirga commented:
Oh wait @karthikscale3 there’s some failures in CI

lmolkova commented:
@karthikscale3 I think you have to regenerate md (`make attribute-registry-generation table-generation` )

karthikscale3 commented:
> make attribute-registry-generation table-generation

Thank you! Regenerated and pushed it back up. Will monitor for any other failures
